src/msg/async/dpdk/DPDK.cc

```diff
@@ -308,7 +308,8 @@ int DPDKDevice::init_port_start()
   }
 
   // TSO is supported starting from DPDK v1.8
-  if (_dev_info.tx_offload_capa & DEV_TX_OFFLOAD_TCP_TSO) {
+  if ((_dev_info.tx_offload_capa & DEV_TX_OFFLOAD_TCP_TSO) &&
```

Comment:
do we have a bug filed against this issue? if yes, could you link to it in the commit message and add it as a comment in the code?

Reply:
@fengchunsong could you reconsider this question? a link to an issue report in https://github.com/spdk/spdk/issues would do.

Reply:
Okay, I will confirm whether to report to centos or dpdk. However, TSO is an optional offload function, which is turned on by default. If the tcp packet is larger than mss, it will fail to be sent, and dpdk will not work. We should not depend on optional features of another component.Therefore, this configuration is required

Reply:
> If the tcp packet is larger than mss, it will fail to be sent

thank you for the detailed explanation. i was not questioning if we need this option. just wanted keep this issue tracked. yeah, if we don't adjust/limit the size of a packet under MSS, this feature is just a recipe to fail the messenger.

once the option is accompanied with the issue link and the MSS limit, it's good to merge.

also, i am unresolving this conversation. because it's not resolved.

Reply:
https://github.com/DPDK/dpdk/commit/a1d0caa92c7fe43cef7a9f6ac4772e0eecd4a011

Reply:
https://github.com/DPDK/dpdk/commit/a1d0caa92c7fe43cef7a9f6ac4772e0eecd4a011
https://github.com/DPDK/dpdk/commit/52e1ece50aaf526b900120283284834b0a59e3ce
https://github.com/DPDK/dpdk/commit/be56d20ff17fdba2568a56e35abf1706cb48bb85
https://github.com/DPDK/dpdk/commit/6d05fc905df216b78e4e3db935927a53a638c6d0

Reply:
Unfortunately, the hns3 and mlx cards I used had tso issues, and they all fixed on version 21.